cmd/template/cluster/provider/common.go

```diff
@@ -51,3 +64,80 @@ func newCAPIV1Alpha3ClusterCR(config ClusterCRsConfig, infrastructureRef *corev1
 
 	return cluster
 }
+
+func runMutation(ctx context.Context, client k8sclient.Interface, input []byte) ([]byte, error) {
+	obj := &unstructured.Unstructured{}
+	dec := apiyaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+	_, gvk, err := dec.Decode(input, nil, obj)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	// Create a DiscoveryClient to get the GVR.
+	dc, err := discovery.NewDiscoveryClientForConfig(client.RESTConfig())
```

Comment:
Should this be created once and passed in rather than creating it for each object?

Reply:
I think I've seen one of these functions cause rate limiting because it makes several requests when it's constructed.